app/portainer/views/account/create-user-access-token/create-user-access-token.controller.js

```diff
@@ -0,0 +1,39 @@
+export default class CreateUserAccessTokenController {
+  /* @ngInject */
+  constructor($async, $state, $analytics, Authentication, UserService, Notifications) {
+    this.$async = $async;
+    this.$state = $state;
+    this.$analytics = $analytics;
+    this.Authentication = Authentication;
+    this.UserService = UserService;
+    this.Notifications = Notifications;
+
+    this.onSubmit = this.onSubmit.bind(this);
+    this.onSuccess = this.onSuccess.bind(this);
+    this.onError = this.onError.bind(this);
+  }
+
+  async onSubmit(description) {
+    const accessToken = await this.UserService.createAccessToken(this.state.userId, description);
```

Comment:
try/catch?

Reply:
ah, ok. you moved this to the form. great. ($state.go is possible from react component)

Reply:
interesting; didnt know that. But im assuming typescript will throw some errors right

Reply:
you can use https://github.com/ui-router/react anywhere in react components. 
so
```
import {useRouter} from '@ui-router/react`;

... inside component
const router = useRouter();
router.stateService.go(stateName)
```

https://ui-router.github.io/react/docs/latest/modules/_hooks_userouter_.html#userouter